oauth2client/transport/httplib2.py

```diff
@@ -44,32 +44,31 @@ def get_http_object(*args, **kwargs):
     return httplib2.Http(*args, **kwargs)
 
 
-def inject_credentials(credentials, http, refresh_status_codes):
-    """Prepares an HTTP object's request method for auth.
-
-    Wraps HTTP requests with logic to catch auth failures (typically
-    identified via a 401 status code). In the event of failure, tries
-    to refresh the token used and then retry the original request.
-
-    Args:
-        credentials: Credentials, the credentials used to identify
-                     the authenticated user.
-        http: httplib2.Http, an http object to be used to make
-              auth requests.
+class AuthorizedHttp(object):
+    """A httplib2.Http-like object that provides credentials to requests.
+
+    A new class is used because you can't create a new OAuth subclass of
+    httplib2.Authentication because it never gets passed the absolute URI,
+    which is needed for signing. So instead we have to overload 'request'
+    and add in the Authorization header and then calls the original
+    version of :func:`request`.
     """
-    orig_request_method = http.request
+    def __init__(self, http, credentials, refresh_status_codes):
+        self.http = http
+        self.credentials = credentials
+        self.refresh_status_codes = refresh_status_codes
 
-    # The closure that will replace 'httplib2.Http.request'.
-    def new_request(uri, method='GET', body=None, headers=None,
-                    redirections=httplib2.DEFAULT_MAX_REDIRECTS,
-                    connection_type=None, _credential_refresh_attempt=0):
+    def request(self, uri, method='GET', body=None, headers=None,
+                redirections=httplib2.DEFAULT_MAX_REDIRECTS,
+                _credential_refresh_attempt=0, **kwargs):
```

Comment:
These aren't in the [signature][1] of the base class. I suppose `kwargs` is meant to cover `connection_type`.

Wish there was a way to avoid adding this extra kwarg

[1]: https://github.com/httplib2/httplib2/blob/5ccd260911e2db802fa97498ae92558fac8c8057/python2/httplib2/__init__.py#L1467